src/HasMedia/MediaCollection.php

```diff
@@ -29,28 +29,22 @@ class MediaCollection {
 	 * @param string $name
 	 */
 	public function __construct( string $name ) {
-		$this->name = $name;
-		$this->disk = config( 'media-collections.public_disk', 'media' );
-	}
 
-	/**
-	 * @param string $name
-	 *
-	 * @return MediaCollection
-	 */
-	public static function create( string $name ): self {
-		return new static( $name );
+        parent::__construct($name);
+
+		$this->diskName = config( 'media-collections.public_disk', 'media' );
 	}
 
+
 	/**
 	 * Specify a disk where to store this collection
 	 *
 	 * @param $disk
-	 *
-	 * @return $this
+     * @deprecated deprecated since version 3.0
```

Comment:
Why is here something deprecated?

Reply:
Because method disk should be deleted in future versions. New Spatie Library comes with built-in method useDisk, which do the same as our disk method. However, I would temporarily keep the disk method to ensure backward compatibility.